src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -331,8 +329,6 @@ private static void WriteTo(Metadata? metadata, ObjectWriter writer, Cancellatio
                 return false;
             }
 
-            context.AddResource(storage);
```

Comment:
this is a similar issue.  walking through all the ITempStorage code, none of the types that we are serializing here which dump themselves into temp-storage ever get disposed.  so there's no need to keep this alive either.